react/FilePicker/FilePickerBodyItem.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+import { models } from 'cozy-client'
+import ListItem from '../MuiCozyTheme/ListItem'
+import ListItemIcon from '../MuiCozyTheme/ListItemIcon'
+import ListItemText from '../ListItemText'
+import Divider from '../MuiCozyTheme/Divider'
+import Icon from '../Icon'
+import FileTypeText from '../Icons/FileTypeText'
+import FileTypeFolder from '../Icons/FileTypeFolder'
+import Checkbox from '../Checkbox'
+import Radio from '../Radio'
+
+const {
+  file: { isDirectory, isFile }
+} = models
+
+const FilePickerBodyItem = ({
+  file,
+  isAccept,
+  multiple,
+  handleChoiceClick,
+  handleListItemClick,
+  onChangeRadio,
+  filesIdsSelected,
+  hasDivider
+}) => {
+  const hasFileChoice = isAccept.file && isFile(file)
+  const hasFolderChoice = isAccept.folder && isDirectory(file)
+
+  return (
+    <>
+      <ListItem button className={cx('u-p-0')}>
+        <div
+          data-testid="choice-onclick"
+          className={cx('u-ph-half')}
+          onClick={
+            hasFileChoice || hasFolderChoice
+              ? handleChoiceClick(file)
+              : undefined
+          }
+        >
+          {multiple ? (
+            <Checkbox
+              data-testid="checkbox-btn"
+              onChange={() => {
+                // handled by onClick on the container
+              }}
+              checked={filesIdsSelected.includes(file._id)}
+              value={file._id}
+              className={cx('u-ph-half', {
+                'u-o-100': hasFileChoice || hasFolderChoice,
+                'u-o-0': !(hasFileChoice || hasFolderChoice)
+              })}
+              disabled={!hasFileChoice && !hasFolderChoice}
+            />
+          ) : (
+            <Radio
+              data-testid="radio-btn"
+              onChange={onChangeRadio}
```

Comment:
Seems like you can ignore this one too like for the Checkbox, can't you?


If this is the case, then you'll have exactly the same code for the radio and the checkbox (only the testid differs). Which means that you can simplify the code by doing something like this :
```jsx
const Child = multiple ? Checkbox : Radio

return (
  <Child
      data-testid="choice-btn"
      onChange={() => {
        // handled by onClick on the container
      }}
      checked={filesIdsSelected.includes(file._id)}
      value={file._id}
      className={cx('u-ph-half', {
        'u-o-100': hasFileChoice || hasFolderChoice,
        'u-o-0': !(hasFileChoice || hasFolderChoice)
      })}
      disabled={!hasFileChoice && !hasFolderChoice}
  />
)
```